os/tools/mksyscall.c

```diff
@@ -487,11 +489,99 @@ static void generate_stub(int nparms)
 		fprintf(stream, ");\n  return 0;\n}\n\n");
 	else
 		fprintf(stream, ");\n}\n\n");
+	stub_close(stream);
+}
+	else{
```

Comment:
Almost same code is repeated in if and else parts. 
Please restructure the changes so that common code is not repeated.